src/Application/Migrations/24_function_ticket_reset.sql

```diff
@@ -0,0 +1,65 @@
+BEGIN;
+
+--param_handle_id is used for the log entry
+CREATE OR REPLACE FUNCTION ticket_reset(param_ticket_id bigint, param_handle_id bigint)
+  RETURNS void AS
+$$
+DECLARE
+  ticket tbl_ticket%rowtype;
+  inital_state enum_ticket_state;
+BEGIN
+  --first get the full given ticket
+  SELECT * INTO ticket FROM tbl_ticket where id = param_ticket_id;
+
+  IF NOT FOUND THEN
+    RAISE WARNING 'ticket with id % does not exist', param_ticket_id;
+    RETURN;
+  END IF;
+
+  --reset the given ticket to inital state
+  inital_state := ticket_state_initial(ticket.project_id, ticket.ticket_type);
+  IF inital_state <> ticket.ticket_state THEN
+    INSERT INTO tbl_log (
+      ticket_id, from_state, to_state, handle_id, event
+      ) VALUES (ticket.id, ticket.ticket_state, inital_state, param_handle_id, 'Action.reset');
+    UPDATE tbl_ticket SET ticket_state=inital_state WHERE id = param_ticket_id;
+  END IF;
+
+  FOR ticket IN
```

Comment:
The comment describing what is going on here was nice, it didn't have to vanish, only get updated. (same below)

Reply:
The plural of child is children ;)